### ISSUE TITLE
Option to control spell-check frequency (e.g. only on file save)

### ISSUE THREAD

**Issue by zertrin** [valentjn/vscode-ltex#142]
**Is your feature request related to a problem? Please describe.**
While writing a large PhD thesis, the time for spell-checking is quite long and the spell-check tends to be constantly repeated while typing sentences, pausing for thinking a bit, then continue writing some more and so on. This repeated spell-checking after typing a few words results in my laptop's CPU and fans to be constantly under heavy usage. Even by increasing the sentence cache, my document is likely too long to be fully cached and thus probably undergoes constant full spell-checking which takes time, CPU and fan usage.

**Describe the solution you'd like**
I don't really need the spell-checker to check the document after small pause in my writing session.

It would be sufficient to launch spell-checking on document save only 
_(which I tend to do after finishing a full paragraph since it triggers LaTeX PDF building which I also don't need to occur after every few words)_.

So the feature request would be to add a setting to vscode-ltex where the user can choose the frequency of spell-checking, for example between `automatic` (current default), `on save` or `manual` (user would need to trigger it from the _"check current document"_ command).

**Describe alternatives you've considered**
If this cannot be implemented, I might have to resort to disabling the extension while in a writing session, and manually re-enable it from time to time to do the spell-checking in batch. But that's poor UX from my point of view.

**Additional context**
n/a

**Comment by valentjn** [comment]
How large is your file (bytes)? In general, if it takes longer than a few seconds, then it's probably a bug (as you noticed, probably #134), especially if you increase the size of the sentence cache. How much did you increase it?

**Comment by zertrin** [comment]
> How large is your file (bytes)?

About 80,000 bytes currently.

> In general, if it takes longer than a few seconds, then it's probably a bug (as you noticed, probably #134),

I agree. Yet, even then, I believe that this feature request could be useful as it enables the user to temporarily mitigate the effects of bugs (not only the current one, but any future bug causing performance issue).

> especially if you increase the size of the sentence cache. How much did you increase it?

I increased the cache size to 10,000 (default being 2,000). It is not very clear what is a reasonable range for this parameter. As a user, it's hard to judge what would be "too much". Can I set it to 100,000? 1,000,000? I guess the answer depends on available memory, but at the same time it's not that easy to measure...
